docs/features/logging.md

```diff
@@ -32,9 +42,63 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env)
 }
 ```
 
-If you have multiple middleware components configured, it is advised to put the `ExceptionHandlingMiddleware` as the first one.  By doing so, unhandled exceptions that might occur in other middleware components will also be logged by the `ExceptionHandlingMiddleware` component.
+If you have multiple middleware components configured, it is advised to put the `ExceptionHandlingMiddleware` as the first one.
+By doing so, unhandled exceptions that might occur in other middleware components will also be logged by the `ExceptionHandlingMiddleware` component.
+
+## Logging incoming requests
+
+The `RequestTrackingMiddleware` class can be added to the <span>ASP.NET</span> Core pipeline to log all received HTTP requests.
+The requests by this middleware component are logged through the `ILogger` implementations that are configured inside the project.
+
+The HTTP request headers and request body are by default logged.
+
+**Example**
+
+`HTTP Request POST http://localhost:5000/weatherforecast completed with 200 in 00:00:00.0191554 at 03/23/2020 10:12:55 +00:00 - (Context: [Content-Type, application/json], [Body, {"today":"cloudy"}])`
+
+**Usage**
+
+To use this middleware, add the following line of code in the `Startup.Configure` method:
+
+```csharp
+public void Configure(IApplicationBuilder app, IWebHostEvironment env)
+{
+    app.UseMiddleware<Arcus.WebApi.Logging.RequestTrackingMiddleware>();
+
+    ...
+    app.UseMvc();
+}
+```
+
+**Configuration**
 
-To get the exceptions logged in Azure Application Insights, configure logging like this when building the `IWebHost` in `Program.cs`:
+Optionally, one can inherit from this middleware component and override the default request header and body extraction to discared one or both.
+Following example discards the request headers:
+
+```csharp
+public class NoRequestHeadersRequestTrackingMiddleware : RequestTrackingMiddleware
+{
+    public NoRequestHeadersRequestTrackingMiddleware(
+        RequestDelegate next,
+        ILogger<RequestTrackingMiddleware> logger) 
+        : base(next, logger)
+        {
+        }
+
+    protected override IDictionary<string, object> ExtractRequestHeaders(IHeaderDictionary requestHeaders)
+    {
+        return new Dictionary<string, object>();
+    }
+}
+```
+
+So the resulting log message becomes:
+
+`HTTP Request POST http://localhost:5000/weatherforecast completed with 200 in 00:00:00.0191554 at 03/23/2020 10:12:55 +00:00 - (Context: [Content-Type, application/json])`
+
+## Application Insights
```

Comment:
Frankly, I think we should recommend using Serilog as well since the requests will be tracked as "traces" when using the current suggestion here.

Would you mind adding a second option?

Reply:
I changed it to a Serlog example.